dali/operators/random/uniform_distribution.h

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_RANDOM_UNIFORM_DISTRIBUTION_H_
+#define DALI_OPERATORS_RANDOM_UNIFORM_DISTRIBUTION_H_
+
+#include <vector>
+#include "dali/operators/random/rng_base.h"
+#include "dali/pipeline/operator/arg_helper.h"
+#include "dali/core/dev_buffer.h"
+
+#define DALI_UNIFORM_DIST_TYPES (uint8_t, int8_t, uint16_t, int16_t, uint32_t, int32_t, uint64_t, \
+                                 int64_t, float16, float, double)
+
+namespace dali {
+
+template <typename T>
+class uniform_int_values_dist {
+ public:
```

Comment:
Could you add docs with description, what this functor is for?

Reply:
done